client/blocks/README.md

```diff
@@ -0,0 +1,39 @@
+
+This file is meant to document why we chose to forego the Blocks API at the time, and what the downsides and benefits of that approach are.
+
+## What were the issues we encountered?
+
+-   3DS payments didn't work with saved cards and Payment Requests because some scripts weren't loaded in those contexts.
+-   The Payment Request Dialog [didn't close during 3DS validation](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1467#discussion_r623877272).
+
+Both of these have since been addressed in WooCommerce Blocks.
+
+## The Downsides of Using the AJAX API
+
+We're unable to use the framework provided by WooCommerce Blocks during checkout with Payment Request Buttons, and we miss out on many things Blocks provides by default, such as a responsive UI, improved error reporting, and more.
+
+More specifically:
+
+-   We have to manually make the UI responsive, and some parts we simply _can't_ make responsive.
+    -   This refers to things like blocking the UI during the payment process, starting and stopping loading indicators, and general UX improvements in that vein.
+-   The checkout process requires many more network requests, slowing the checkout process down a significant amount.
+-   Some errors returned from our AJAX API are HTML notices. That HTML is not displayed correctly when the error is shown in the Cart or Checkout blocks.
+
+## The Benefits of Using the AJAX API
+
+There are several benefits to using our pre-existing AJAX APIs:
+
+-   Backwards compatibility with older version of WooCommerce Blocks that were lacking support for crucial features related to Payment Requests.
+    -   This is no longer true as of WooCommerce Blocks v3.0.0.
+-   Code re-use; we already have well tested code in the AJAX API that's quite complicated to test, e.g. numerous normalisation functions.
+    -   Another aspect to this point is that the Shortcode flow in `assets/js` is currently not built in the same system as the blocks integration; to maximise code re-use between the Shortcode- and Block-based checkout we'd have to make some changes to the Shortcode checkout, making the changes required that much bigger.
+-   We don't need to re-implement the payment flow using the Blocks API.
+    -   The WooCommerce Blocks API has been improved significantly since we started developing this integration, so this is no longer as big of a deal as it was.
+-   We get more direct control over how the Payment Request flow is managed.
+
+Another factor in this decision was that during development some parts of the WooCommerce Blocks API were unable to properly handle the required flow for the Payment Request Buttons properly, especially when it came to 3DS payments. This has since been fixed, and anyone else should consider using the WooCommerce Blocks APIs.
```

Comment:
```suggestion
Another factor in this decision was that during development, some parts of the WooCommerce Blocks API were unable to properly handle the required flow for the Payment Request Buttons, especially when it came to 3DS payments. This has since been fixed, and anyone else should consider using the WooCommerce Blocks APIs.
```

Feel free to ignore this, I added a comma on the first sentence and removed the second instance of `properly` at the beginning.

Reply:
Fixed in b8dfb9fd21c8700c7f39281cde0f959c5474a1ee